### ISSUE TITLE
Cannot add extra public nics to VPC VR

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#5890]
when acquire a public IP in additional public Ip range in VPC, and create port forwarding rule, it failed with following error logs

```
2022-01-24 10:18:12,167 ERROR [c.c.v.VmWorkJobHandlerProxy] (Work-Job-Executor-29:ctx-1244b16f job-191/job-192 ctx-0839cf31) (logid:5e3e1e3c) Invocation exception, caused by: com.cloud.utils.exception.CloudRuntimeException: A NIC already exists for VM:r-26-VM in network: 2cd53639-233b-4bc2-b5bd-19751834ab9e
2022-01-24 10:18:12,168 INFO  [c.c.v.VmWorkJobHandlerProxy] (Work-Job-Executor-29:ctx-1244b16f job-191/job-192 ctx-0839cf31) (logid:5e3e1e3c) Rethrow exception com.cloud.utils.exception.CloudRuntimeException: A NIC already exists for VM:r-26-VM in network: 2cd53639-233b-4bc2-b5bd-19751834ab9e
2022-01-24 10:18:12,169 DEBUG [c.c.v.VmWorkJobDispatcher] (Work-Job-Executor-29:ctx-1244b16f job-191/job-192) (logid:5e3e1e3c) Done with run of VM work job: com.cloud.vm.VmWorkAddVmToNetwork for VM 26, job origin: 191

com.cloud.utils.exception.CloudRuntimeException: A NIC already exists for VM:r-26-VM in network: 2cd53639-233b-4bc2-b5bd-19751834ab9e
        at com.cloud.vm.VirtualMachineManagerImpl.checkIfNetworkExistsForVM(VirtualMachineManagerImpl.java:3851)
        at com.cloud.vm.VirtualMachineManagerImpl.orchestrateAddVmToNetwork(VirtualMachineManagerImpl.java:3860)
        at com.cloud.vm.VirtualMachineManagerImpl.orchestrateAddVmToNetwork(VirtualMachineManagerImpl.java:5363)
```

(2cd53639-233b-4bc2-b5bd-19751834ab9e is the uuid of Public network with id=200)

- steps to reproduce the issue

(1) create VPC
(2) add additional public ip range
(3) acquire a public IP in new ip range
(4) create port forwarding rule (or enable static nat, create LB rule)

expected result: succeed
actual result: failed

- **it can be fixed by reverting #5658** 

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
main branch
should exist in 4.16 as well
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
@sureshanaparti @DaanHoogland 
found a regress issue in main (should in 4.16 as well). add it to 4.16

**Comment by sureshanaparti** [comment]
Fixed in #5896
